Fix minor typos in string manipulation section (#2123)
nushell/nushell.github.io: book/working_with_strings.md

['foo', 'bar'] | str replace -r '$' '~/'# foo~/, bar~/
```

> If you want to get one string out at the end then `str join` is your friend:

Reviewer: Maybe reword it?
```suggestion
If you want to combine a list into a string, then `str join` is your friend:
```
